packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts

```diff
@@ -81,6 +81,190 @@ describe('HasMany relation', () => {
+      expect(customer.orders).to.matchEach((v: Partial<Order>) => {
+        expect(v).to.deepEqual({
+          id: undefined,
+          description: order.description,
+          customerId: undefined,
+        });
+      });
+    });
+  });
+
+  it('can include the related model when the foreign key is disabled in filter', async () => {
+    const order = await customerOrderRepo.create({
+      description: 'an order desc',
+    });
+    const customer = await customerRepo.findById(existingCustomerId, {
+      include: [
+        {
+          relation: 'orders',
+          scope: {
+            fields: {
+              customerId: false,
+              description: true,
+            },
+          },
+        },
+      ],
+    });
+
+    withProtoCheck(false, () => {
+      expect(customer.orders).length(1);
+      expect(customer.orders).to.matchEach((v: Partial<Order>) => {
+        expect(v).to.deepEqual({
+          id: undefined,
+          description: order.description,
+          customerId: undefined,
+        });
+      });
+    });
+  });
+
+  it('can include the related model when only the foreign key is disabled in filter', async () => {
+    const order = await customerOrderRepo.create({
+      description: 'an order desc',
+    });
+    const customer = await customerRepo.findById(existingCustomerId, {
+      include: [
+        {
+          relation: 'orders',
+          scope: {
+            fields: {
+              customerId: false,
+            },
+          },
+        },
+      ],
+    });
+
+    withProtoCheck(false, () => {
+      expect(customer.orders).length(1);
+      expect(customer.orders).to.matchEach((v: Partial<Order>) => {
+        expect(v).to.deepEqual({
+          id: order.id,
+          description: order.description,
+          customerId: undefined,
+        });
+      });
+    });
+  });
+
+  it('preserves the foreign key value when set in filter', async () => {
+    const order = await customerOrderRepo.create({
+      description: 'an order desc',
+    });
+    const customer = await customerRepo.findById(existingCustomerId, {
+      include: [
+        {
+          relation: 'orders',
+          scope: {
+            fields: {
+              customerId: true,
+              description: true,
+            },
+          },
+        },
+      ],
+    });
+
+    withProtoCheck(false, () => {
+      expect(customer.orders).length(1);
+      expect(customer.orders).to.matchEach((v: Partial<Order>) => {
+        expect(v).to.deepEqual({
+          id: undefined,
+          description: order.description,
+          customerId: order.customerId,
+        });
+      });
+    });
+  });
+
+  it('includes only the fields set in filter', async () => {
+    await customerOrderRepo.create({
+      description: 'an order desc',
+    });
+    const customer = await customerRepo.findById(existingCustomerId, {
+      include: [
+        {
+          relation: 'orders',
+          scope: {
+            fields: {},
```

Comment:
If the fields is `{}`, I'd expect that all the properties would be included. Same for the test at line 486.

Reply:
It is the current behavior for both lb4 and juggler filters. I only verify here that it was not affected.

Reply:
oh wow I didn't know this. Thanks for verifying.